itests/wdpost_dispute_test.go

```diff
@@ -285,36 +286,34 @@ func TestWindowPostDisputeFails(t *testing.T) {
 
 	// Wait until a proof has been submitted.
 	var targetDeadline uint64
+	var dInfo *dline.Info
 waitForProof:
 	for {
+		head, err := client.ChainHead(ctx)
+		require.NoError(t, err)
+
+		di, err := client.StateMinerProvingDeadline(ctx, maddr, head.Key())
+		require.NoError(t, err)
+
 		deadlines, err := client.StateMinerDeadlines(ctx, maddr, types.EmptyTSK)
 		require.NoError(t, err)
 		for dlIdx, dl := range deadlines {
 			nonEmpty, err := dl.PostSubmissions.IsEmpty()
 			require.NoError(t, err)
-			if nonEmpty {
+			if nonEmpty && di.Index == uint64(dlIdx) {
```

Comment:
what's with the naming of this variable? shouldn't it be `isEmpty`?

Reply:
I think also, you could flip this and do `di.Index == uint64(dlIdx)` in an outer `if` and the `IsEmpty` on an inner to avoid the unnecessary call and err check on `IsEmpty`